base/array.jl

```diff
@@ -916,9 +916,11 @@ function push!(a::Array{Any,1}, @nospecialize item)
 end
 
 """
-    append!(collection, collection2) -> collection.
+    append!(a::Vector, items) -> collection.
```

Comment:
Why this change?

Reply:
`a` has to be a vector for `append!` to work. Of course, `items` could stay `collection`, but this wording is consistent with the other docstrings. Perhaps `collection` would be clearer, though, as items may be interpreted as several seperate items and `append!` expects a collection of items...

```julia
julia> append!([1 2 3], 1)
ERROR: MethodError: no method matching append!(::Array{Int64,2}, ::Int64)

You might have used a 2d row vector where a 1d column vector was required.
Note the difference between 1d column vector [1,2,3] and 2d row vector [1 2 3].
You can convert to a column vector with the vec() function.
Closest candidates are:
  append!(::Array{T,1} where T, ::Any) at array.jl:637

julia> append!([1,2,3], 1)
4-element Array{Int64,1}:
 1
 2
 3
 1
```

Reply:
This change should go to a separate commit or PR, and it should use `AbstractVector`. I also wonder whether this isn't on purpose, since non-`AbstractVector` collections defined in packages could define methods and it would be silly to repeat the information in another docstring.